govuk-one-login/onboarding-product-page
# SSE-3210: Jira integration for Register Your Interest Form

Dependent on: https://github.com/govuk-one-login/onboarding-self-service-infra/pull/21

## Why

Enables an integration requested by the onboarding team.

Currently we take data from the register your interest form and send it to a Google Sheet for staff to work on.
We are looking to automate aspects of that process by sending the data directly into 
Jira instead.

## What

[Screencast from 12-03-24 14:22:31.webm](https://github.com/govuk-one-login/onboarding-product-page/assets/3694062/6812b956-ea94-4697-af0f-5e4a7bbe1d9e)

This PR:
- Adds a new Jira service to handle posting to create issues
- Creates a stub to enable integration testing
- Masks the real and stub interface behind a single class, sets the toggle between functions as an Env Var
- Adds the integration to the register your interest control with a feature flag
- Adds a feature flag defaulted to ON for the google sheets feature

It should look like this:


## How to Test
- Whilst logged into Jira [visit the Atlassian Dashboard](https://id.atlassian.com/manage-profile/security) and register yourself a personal API token.
- Copy across the new .env.example into your .env and populate with your work email as username and the token you just created as the credential
- Populate the env with JIRA_PROJECT_KEY=APIT and JIRA_ISSUE_TYPE=10907)
- Spin up the local application and processed to fill out the reigster your interest form
- Submit and check a value as appeared on the [test board](https://govukverify.atlassian.net/jira/software/projects/APIT/boards/510)
- Turn the feature flag off in the ENV for Jira, re-test and check no calls / board values appear
- Turn off the feature flag for google sheets and check from the logs that the stub has not been called

After that review code normally
